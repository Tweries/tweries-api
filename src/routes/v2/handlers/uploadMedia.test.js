const uploadMedia = require('./uploadMedia');

test('uploadMedia', async () => {
  const mockPost = jest.fn(() => ({ data: { media_id_string: '0x123' } }));

  const options = {
    file: { buffer: Buffer.from('HELLO'), originalname: 'hello.jpg' },
    T: { post: mockPost }
  };

  const result = await uploadMedia(options);

  expect(mockPost.mock.calls[0][1]).toMatchSnapshot();
  expect(mockPost.mock.calls[1][1]).toMatchSnapshot();
  expect(result).toMatchSnapshot();
});
