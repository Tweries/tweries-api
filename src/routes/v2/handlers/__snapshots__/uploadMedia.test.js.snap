// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uploadMedia 1`] = `
Object {
  "media_data": "SEVMTE8=",
}
`;

exports[`uploadMedia 2`] = `
Object {
  "alt_text": Object {
    "text": "hello.jpg",
  },
  "media_id": "0x123",
}
`;

exports[`uploadMedia 3`] = `
Object {
  "mediaIdString": "0x123",
  "media_id_string": "0x123",
}
`;
